### PR TITLE
Feat: Add the Newsletter feature to the website. 

### PR DESCRIPTION
**Pull request by ayushnau**
**GitHub Issue:** #243 

**Summary**:

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**
No

No

### CODE REVIEW

**Review by benjagm — CHANGES_REQUESTED** [review]
Thanks for this PR @ayushnau !

We'll need to do some additional work to make sure we provide the name and email fields alongside with the subscribe button in the landing page like the example provided from AsyncAPI.

Screenshot 2024-02-15 at 11 27 40

In addition, the new page should have all the styling than the other pages of the website including headers and footer... you can see a similar page in "https://json-schema.org/implementations" and the equivalent page in ASyncAPI site:

https://www.asyncapi.com/newsletter

Is it ok to rename the page subscribe to newsletter?

Could it be possible to avoid uppercase in the name of Newsletter.tsx?

**Comment by ayushnau** [comment]
@benjagm can you elaborate the newsletter point. do you want it for filename or for component name . cause i can see in some places we have used uppercases in filename and also in component name.

**Comment by benjagm** [comment]
> newsletter point

Forget about this one @ayushnau. My bad, this as internal component and uppercase in the first letter is irrelevant.

**Comment by ayushnau** [comment]
ok can you review the pr. here is how it looks.
image

**Comment by ayushnau** [comment]
Also please provide me the api for login.

**Review by benjagm — CHANGES_REQUESTED** [review]
The screenshot I provided was just an example, so it is not necessary to add that black background.

- For the newsletter page: Background should be white, fonts smaller and increase the horizontal padding to at least 8.

- For showing the component in the landing page remember it is going to be shown in this section (make sure the styles work here and also in the newsletter page):

Screenshot 2024-02-19 at 10 58 41

**Comment by ayushnau** [comment]
image
@benjagm

**Comment by ayushnau** [comment]
@benjagm

File: components/Newsletter.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react'
+import classnames from 'classnames'
+
+interface NewsletterFormProps {
+  wrapperClassName?: string
+  className?: string
+}
+const NewsletterForm: React.FC<NewsletterFormProps> = ({ wrapperClassName = '', className = '' }) => {
```
Comment (benjagm) [inline components/Newsletter.tsx:8]: This component is going to be presented in two scenarios:
1. In the newsletter page with while background and black letters. (default)
2. In the landing page call to action banners, with blue background and white letters. (see image)

Screenshot 2024-02-21 at 11 45 31

With that in mind you will need to define an input parameter allowing to define what of those 2 behaviours the component will render.

File: components/Newsletter.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          />
+          <input
+            type='email'
+            name='email'
+            placeholder='Your Email'
+            className='form-input block w-full py-3 text-lg sm:text-lg sm:leading-5  md:flex-1 rounded-md px-5 bg-gray-200'
+            value={email}
+            onChange={(e: any) => setEmail(e.target.value)}
+          />
+          <button
```
Comment (benjagm) [inline components/Newsletter.tsx:51]: For the button styles please use the same fonts, color border than we use in the landing page:

Screenshot 2024-02-21 at 11 43 54

File: pages/index.page.tsx [hunk 3]
```diff
@@ -320,7 +320,7 @@ const Home = (props: any) => {
         <section className='w-full h-[300px] lg:h-[367px] bg-gradient-to-r from-primary from-1.95% to-endBlue clip-both'>
           <div className='lg:w-full mx-auto text-center mt-28 '>
             <h2 className='text-h3mobile lg:text-h3 text-white mb-6'>Start contributing to JSON Schema</h2>
-            <button className='w-[170px] h-[45px] mx-auto rounded border-2 bg-primary text-white font-semibold'><a href='https://github.com/json-schema-org#-contributing-to-json-schema'>Contribute</a></button>
+            <button className='w-[170px] h-[45px] mx-auto rounded border-2 bg-primary text-white font-semibold'><a href='/newsletter'>Subscribe</a></button>
```
Comment (benjagm) [inline pages/index.page.tsx:323]: Wha we need here is to show the component itself, no a button to navigate to the newsletter page. We need here to remove that extra step. Make it is easy for the user to subscribe.

See my first comment regarding how to make the component use different fonts/colours depending on the context.

My expectation is to implement the same approach than asyncapi. They show component in the landing page with white fonts:
https://www.asyncapi.com/en
And they show the component with black fonts in the newsletter page: 
https://www.asyncapi.com/newsletter

File: components/Newsletter.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  //   e.preventDefault()
+  //   // Handle form submission here, such as sending a request to subscribe the user
+  // }
+
+  // const handleFileChange = (e: any) => {
+    
+  // }
+
+  return (
+    <section
```
Comment (benjagm) [inline components/Newsletter.tsx:22]: We are very close with the desktop implementation, however we need to keep working for mobile and small devices displays.

Screenshot 2024-02-21 at 11 39 47

Please see the behaviour of Asyncapi component and adjust accordingly
Screenshot 2024-02-21 at 11 39 35
:

**Review by benjagm — CHANGES_REQUESTED** [review]
You did a lot of progress @ayushnau , Thanks!!

I left some additional change requests to make sure we are covering all the scenarios.

Great progress so far!!! Congrats. We are very close.

**Comment by AyushNautiyalDeveloper** [comment]
image
image
image
image
@benjagm

**Comment by AyushNautiyalDeveloper** [comment]
Pasted ss for newsletter page and landing page both in mobile and web views.

**Review by benjagm — CHANGES_REQUESTED** [review]
@AyushNautiyalDeveloper @ayushnau Are you the same person?

Please check and increase the horizontal paddings or margins in mobile to make it look similar to AsyncAPI:
306595897-588260ae-02d9-4178-977c-627170bafc7f

For desktop: This is what I see
Screenshot 2024-02-23 at 13 38 38

I think something is missing when appearing in the landing page. In addition in this case I think the background of the inputboxes will look better white.

**Comment by ayushnau** [comment]
image
image
made the changes. here is the images for reference.  @benjagm

File: pages/index.page.tsx (hunk 3 above)
Comment (ayushnau) [reply pages/index.page.tsx:324]: added.

**Review by benjagm — APPROVED** [review]
Excelent work @ayushnau!

Just pushed some extra code to make it work with our mailchimp account. Will merge it in the next days!

**Comment by ayushnau** [comment]
ok sure that would be great. @benjagm

**Comment by benjagm** [comment]
@ayushnau Do you mind rebasing and solving the merge conflicts? We'll merge it.

**Review by benjagm — CHANGES_REQUESTED** [review]
Can we please avoid the changes in package.json and yarn.lock? I think all those changes are not necessary.

**Comment by benjagm** [comment]
Can you please change the target branch to https://github.com/json-schema-org/website/tree/web-release-3?

Reference docs: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

**Comment by DhairyaMajmudar** [comment]
@ayushnau do you need any help for the pr, feel free to ask : )

**Comment by benjagm** [comment]
Can you please change the target branch to https://github.com/json-schema-org/website/tree/web-release-3?

Reference docs: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

**Comment by benjagm** [comment]
Are we closing this in favour of #489 ?

**Comment by AyushNautiyalDeveloper** [comment]
> Are we closing this in favour of #489 ?

yes.

**Comment by benjagm** [comment]
Closed in favour of #489

**Pull request closed without merging**